WordPress/Classes/ViewRelated/System/Notices/NoticePresenter.swift

```diff
@@ -0,0 +1,198 @@
+            return
+        }
+
+        let noticeView = NoticeView(notice: notice)
+        noticeView.translatesAutoresizingMaskIntoConstraints = false
+
+        let noticeContainerView = NoticeContainerView(noticeView: noticeView)
+        addNoticeContainerToPresentingViewController(noticeContainerView)
+
+        let bottomConstraint = makeBottomConstraintForNoticeContainer(noticeContainerView)
+
+        NSLayoutConstraint.activate([
+            noticeContainerView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            noticeContainerView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            bottomConstraint
+        ])
+
+        let fromState = {
+            noticeView.alpha = WPAlphaZero
+            bottomConstraint.constant = self.offscreenBottomOffset
+
+            view.layoutIfNeeded()
+        }
+
+        let toState = {
+            noticeView.alpha = WPAlphaFull
+            bottomConstraint.constant = 0
+
+            view.layoutIfNeeded()
+        }
+
+        let dismiss = {
+            guard noticeContainerView.superview != nil else {
+                return
+            }
+
+            self.animatePresentation(fromState: {}, toState: fromState, completion: {
+                self.dismiss()
+            })
+        }
+
+        noticeView.dismissHandler = dismiss
+
+        animatePresentation(fromState: fromState, toState: toState, completion: {
+            DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + Animations.dismissDelay, execute: dismiss)
+        })
+    }
+
+    private func dismiss() {
+        ActionDispatcher.dispatch(NoticeAction.dismiss)
+    }
+
+    private func addNoticeContainerToPresentingViewController(_ noticeContainer: UIView) {
+        if let tabBarController = presentingViewController as? UITabBarController {
+            tabBarController.view.insertSubview(noticeContainer, belowSubview: tabBarController.tabBar)
+        } else {
+            presentingViewController.view.addSubview(noticeContainer)
+        }
+    }
+
+    private func makeBottomConstraintForNoticeContainer(_ container: UIView) -> NSLayoutConstraint {
+        if let tabBarController = presentingViewController as? UITabBarController {
+            return container.bottomAnchor.constraint(equalTo: tabBarController.tabBar.topAnchor)
+        }
+
+        // Force unwrapping, as the calling method has already guarded against a nil view
+        return container.bottomAnchor.constraint(equalTo: presentingViewController.view!.bottomAnchor)
+    }
+
+    private var offscreenBottomOffset: CGFloat {
+        if let tabBarController = presentingViewController as? UITabBarController {
+            return tabBarController.tabBar.bounds.height
+        } else {
+            return 0
+        }
+    }
+
+    typealias AnimationBlock = () -> Void
+
+    private func animatePresentation(fromState: AnimationBlock, toState: @escaping AnimationBlock, completion: @escaping AnimationBlock) {
```

Comment:
This felt familiar, so I went searching. Did you check the `Animator` class? That also reminded me of `NoticeAnimator`, so making a note that we might want to update those "notices" to use the new Notices UI 😁